pkg/machine/hyperv/vsock/vsock.go

```diff
@@ -7,10 +7,13 @@ import (
 	"fmt"
 	"io"
 	"net"
+	"os"
+	"os/exec"
 	"strings"
 
 	"github.com/Microsoft/go-winio"
 	"github.com/containers/podman/v5/pkg/machine/sockets"
+	"github.com/containers/podman/v5/pkg/machine/wsl/wutil"
```

Comment:
do we do this in other places in WSL? I dont like mixing providers by calling into each other when possible.  would you consider `pkg/machine/windows` instead (apple does this)